python_anvil/api.py

```diff
@@ -291,11 +310,11 @@ def create_etch_packet(
             raise ValueError(
                 "`payload` must be a valid CreateEtchPacket instance or dict"
             )
-        return self.mutate(
-            mutation,
-            variables=mutation.create_payload().dict(by_alias=True, exclude_none=True),
-            **kwargs,
-        )
+
+        files = get_multipart_payload(mutation)
+
+        res = self.mutate_multipart(files, **kwargs)
```

Comment:
A few things:
- This looks like it will always call `mutatate_multipart`, even if there are no binary uploads, and therfore no reason to use `multi-part`. Is that right? Maybe only use `multi-part` when necessary?
- It feels like this decision to use multipart or not, and the few lines hre of what to do in that case are better abstracted away inside the internals of `self.mutate` somehow. That way anything that is uploading binary in a mutation will just have that stuff handled implicitly.